joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -44,8 +44,18 @@ trait AstCreatorHelper { this: AstCreator =>
           case _ =>
         // Do nothing
         pni
-      case Success(nodeReferenceId) => parserNodeCache(nodeReferenceId)
-
+      case Success(nodeReferenceId) =>
+        // Get the parser node info from the cache using the node reference ID
+        val parserNodeInfo = parserNodeCache.get(nodeReferenceId)
+
+        // If the parser node info exists in the cache, return it
+        parserNodeInfo match
+          case Some(value) => value
```

Comment:
In general, I'd prefer more descriptive names than `value`, but not worth delaying this PR on alone